src/rbr/RoutingTableGenerator.java

```diff
@@ -89,25 +89,23 @@ private int maxOfRegions() {
     private void PrintRegions(Vertex router, BufferedWriter bw) throws IOException {
         bw.append("\n -- Router " + router.name() + "\n");
         bw.append("(");
-
-        for (int regionIndex = 0; regionIndex < regionsForVertex.get(router).size(); regionIndex++) {
-            Region currentRegion = regionsForVertex.get(router).get(regionIndex);
+        for (Region currentRegion : regionsForVertex.get(router)) {
             int Xmin = currentRegion.box().min(0);
             int Ymin = currentRegion.box().min(1);
             int Xmax = currentRegion.box().max(0);
             int Ymax = currentRegion.box().max(1);
 
             String region = "(\""
-                    + opToBinary(regionsForVertex.get(router).get(regionIndex).inputPorts())
+                    + opToBinary(currentRegion.inputPorts())
                     + intToBinary(Xmin, bitPerCoordinate)
                     + intToBinary(Ymin, bitPerCoordinate)
                     + intToBinary(Xmax, bitPerCoordinate)
                     + intToBinary(Ymax, bitPerCoordinate)
-                    + opToBinary(regionsForVertex.get(router).get(regionIndex).outputPorts())
+                    + opToBinary(currentRegion.outputPorts())
                     + "\")";
 
             bw.append(region);
-            if(regionIndex < maxOfRegions - 1)
+            if (regionsForVertex.get(router).indexOf(currentRegion) < maxOfRegions - 1)
```

Comment:
Unfortunately the code rely on the order of the iteration (the last region has to be the last of the loop) and using a foreach (or iterator) in that situation is not recommended. I'm afraid you'll have to drop that mod also. :worried: